modules/telco-observability-key-performance-metrics.adoc

```diff
@@ -0,0 +1,141 @@
+
+|===
+
+.API latency by verb
+[options="header"]
+|===
+
+|Metric|Query
+
+|`GET`
+|`histogram_quantile (0.99, sum by (le,managed_cluster) (sum_over_time(apiserver_request_duration_seconds_bucket{apiserver=~"kube-apiserver\|openshift-apiserver", verb="GET"}[60m])))`
+
+|`PATCH`
+|`histogram_quantile (0.99, sum by (le,managed_cluster) (sum_over_time(apiserver_request_duration_seconds_bucket{apiserver="kube-apiserver\|openshift-apiserver", verb="PATCH"}[60m])))`
+
+|`POST`
+|`histogram_quantile (0.99, sum by (le,managed_cluster) (sum_over_time(apiserver_request_duration_seconds_bucket{apiserver="kube-apiserver\|openshift-apiserver", verb="POST"}[60m])))`
+
+|`LIST` 
+|`histogram_quantile (0.99, sum by (le,managed_cluster) (sum_over_time(apiserver_request_duration_seconds_bucket{apiserver="kube-apiserver\|openshift-apiserver", verb="LIST"}[60m])))`
+
+|`PUT`
+|`histogram_quantile (0.99, sum by (le,managed_cluster) (sum_over_time(apiserver_request_duration_seconds_bucket{apiserver="kube-apiserver\|openshift-apiserver", verb="PUT"}[60m])))`
+
+|`DELETE`
+|`histogram_quantile (0.99, sum by (le,managed_cluster) (sum_over_time(apiserver_request_duration_seconds_bucket{apiserver="kube-apiserver\|openshift-apiserver", verb="DELETE"}[60m])))`
+
+|Combined
+|`histogram_quantile(0.99, sum by (le,managed_cluster) (sum_over_time(apiserver_request_duration_seconds_bucket{apiserver=~"(openshift-apiserver\|kube-apiserver)", verb!="WATCH"}[60m])))`
+
+|===
+
+.`etcd`
+[options="header"]
+|===
+
+|Metric|Query
+
+|`fsync` 99th percentile latency (per instance)
+|`histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[2m]))`
+
+|`fsync` 99th percentile latency (per cluster)
+|`sum by (managed_cluster) ( histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[60m])))`
+
+|Leader elections
+|`sum(rate(etcd_server_leader_changes_seen_total[1440m]))`
+
+|Network latency
+|`histogram_quantile(0.99, rate(etcd_network_peer_round_trip_time_seconds_bucket[5m]))`
+
+|===
+
+.Operator health
+[options="header"]
+|===
+
+|Metric|Query
+
+|Degraded operators
+|`sum by (managed_cluster, name) (avg_over_time(cluster_operator_conditions{condition="Degraded", name!="version"}[60m]))`
+
+|Total degraded operators per cluster
+|`sum by (managed_cluster) (avg_over_time(cluster_operator_conditions{condition="Degraded", name!="version"}[60m] ))`
+
+|===
+
+== Recommendations for storage of metrics
+
+Out of the box, Prometheus does not back up saved metrics with persistent storage. 
+If you restart the Prometheus pods, all metrics data are lost. 
+You should configure the monitoring stack to use the back-end storage that is available on the platform. 
+To meet the high IO demands of Prometheus you should use local storage.
+
+For Telco core clusters, you can use the `LocalStorage` Operator for persistent storage for Prometheus. 
```

Comment:
```suggestion
For Telco core clusters, you can use the Local Storage Operator for persistent storage for Prometheus. 
```

Reply:
The name of the Operator, afaict, doesn't use backticks. If there's a CR for the Operator, then that would be in backticks for sure, but this reads as the Operator, not the CR for the Operator.

Reply:
Thanks again @skrthomas  - very useful comments